File: asset_leasing/events/item.py
"""Item validate handler - P1-02."""

import frappe
from frappe import _


def validate(doc, method=None):
	if not doc.get("al_is_rental_item"):
		return

	# VAL-P1-06 - mutually exclusive by definition.
	if doc.get("is_fixed_asset"):
		frappe.throw(
			_("An item cannot be both <b>Is Fixed Asset</b> and <b>Is Rental Charge Item</b>."
			  "<br><br>ERPNext books a fixed-asset item on a Sales Invoice as an asset "
			  "<b>disposal</b>. A rental charge item must be a non-stock service item."),
			title=_("Conflicting Item Configuration"),
		)

	if doc.get("is_stock_item"):
		frappe.msgprint(
			_("Rental charge items are normally non-stock service items."),
			indicator="orange", alert=True,
		)

File: asset_leasing/public/js/item.js
// Item - warn early when a rental charge item is configured as a fixed asset.
// Authoritative check is events/item.py.

frappe.ui.form.on('Item', {
	al_is_rental_item(frm) {
		if (frm.doc.al_is_rental_item && frm.doc.is_fixed_asset) {
			frappe.msgprint({
				title: __('Conflicting Item Configuration'),
				indicator: 'red',
				message: __(
					'An item cannot be both a Fixed Asset item and a Rental Charge Item. A rental charge item must be a non-stock service item.'
				),
			});
		}
	},

	is_fixed_asset(frm) {
		if (frm.doc.is_fixed_asset && frm.doc.al_is_rental_item) {
			frm.set_value('al_is_rental_item', 0);
			frappe.show_alert({
				message: __('Is Rental Charge Item cleared — a fixed-asset item cannot bill rent.'),
				indicator: 'orange',
			});
		}
	},
});
